docs/src/main/sphinx/develop/supporting-merge.rst

```diff
@@ -0,0 +1,389 @@
+        ConnectorTableHandle tableHandle)
+
+  This method is called as the engine starts processing a ``MERGE`` statement.
+  The connector must return a ``RowChangeParadigm`` enum instance.  If the
+  connector does not support ``MERGE`` it should throw a ``NOT_SUPPORTED``
+  exception, meaning that ``SQL MERGE`` is not supported by the connector.
+
+* ``getMergeRowIdColumnHandle()``::
+
+    ColumnHandle getMergeRowIdColumnHandle(
+        ConnectorSession session,
+        ConnectorTableHandle tableHandle,
+        MergeDetails mergeDetails)
+
+  This method is called in the early stages of query planning for ``MERGE``
+  statements.  The ColumnHandle returned provides the ``rowId`` used by the
+  connector to identify rows to be merged, as well as any other fields of
+  the row that the connector needs to complete the ``MERGE`` operation.
+
+* ``getWriteRedistributionColumns()``::
+
+    List<ColumnHandle> getWriteRedistributionColumns(
+        ConnectorSession session,
+        ConnectorTableHandle table)
+
+  This method returns a list of ``ColumnHandles`` for table columns that
+  impact write redistribution, e.g., columns that impact partitioning or
+  bucketing.  By default, this method returns an empty list.
+
+* ``beginMerge()``::
+
+    ConnectorMergeTableHandle beginMerge(
+         ConnectorSession session,
+         ConnectorTableHandle tableHandle,
+         MergeDetails mergeDetails)
+
+  As the last step in creating the ``MERGE`` execution plan, the connector's
+  ``beginMerge()`` method is called, passing the ``session``, the
+  ``tableHandle`` and the ``MergeDetails`` object.
+
+  ``beginMerge()`` performs any orchestration needed in the connector to
+  start processing the ``MERGE``.  This orchestration varies from connector
+  to connector.  In the Hive ACID connector, for example, ``beginMerge()``
+  checks that the table is transactional and that all updated columns are
+  writable, and starts a Hive Metastore transaction.
+
+  ``beginMerge()`` returns a ``ConnectorMergeTableHandle`` with any added
+  information the connector needs when the handle is passed back to
+  ``finishMerge()`` and the split generation machinery.  For most
+  connectors, the returned table handle contains at least a flag identifying
+  the table handle as a table handle for a ``MERGE`` operation.
+
+* ``finishMerge()``::
+
+      void finishMerge(
+          ConnectorSession session,
+          ConnectorMergeTableHandle tableHandle,
+          Collection<Slice> fragments)
+
+  During ``MERGE`` processing, the Trino engine accumulates the ``Slice``
+  collections returned by ``ConnectorMergeSink.finish()``.  The engine calls
+  ``finishMerge()``, passing the table handle and that collection of
+  ``Slice`` fragments.  In response, the connector takes appropriate actions
+  to complete the ``MERGE`` operation.  Those actions might include
+  committing the transaction, assuming the connector supports a transaction
+  paradigm.
```

Comment:
Remove "transaction paradigm", since connector's underlying transactions may be totally different from Trino transactions. 

```suggestion
  committing an underlying transaction (if any) or freeing any other resources.
```

Reply:
Changed.